File: GameClient/Assets/Scripts/GameLogic/Login/LoginView.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using UnityEngine.UI;

public class LoginView : ViewBase
{
    private GameObject loginViewPanelPrefab;
    private GameObject loginViewPanel;
    private InputField usernameInputField;
    private InputField passwordInputField;
    private Button loginButton = null;

    public LoginView()
    {
        //Start();
    }

    public override void Start()
    {
        base.Start();

        // 1. 实例化一个 prefab
        // 2. 搜索并持有所需控件
        // 3. 绑定控件的回调

        loginViewPanelPrefab = Resources.Load("Prefabs/UI/LoginView") as GameObject;

        loginViewPanel = Object.Instantiate(loginViewPanelPrefab, UIManager.Instance.UICanvas.transform);
        usernameInputField = loginViewPanel.transform.Find("UsernameLabel/UsernameInputField").GetComponent<InputField>();
        passwordInputField = loginViewPanel.transform.Find("PasswordLabel/PasswordInputField").GetComponent<InputField>();
        loginButton = loginViewPanel.transform.Find("LoginButton").GetComponent<Button>();

        loginButton.onClick.AddListener(OnClickLoginBtn);
    }

    void OnClickLoginBtn()
    {
        // 点击登录
        string username = usernameInputField.text;
        string pwd = passwordInputField.text;

        LoginController.Instance.Login(username, pwd);
    }

    public override void OnDestroy()
    {
        base.OnDestroy();
        Object.DestroyImmediate(loginViewPanel);
    } 

}

File: GameClient/Assets/HotFix/Lua/MVC/Login/LoginView.lua
local loginView = class(require"MVC/Base/BaseView")

function loginView:start( )
	
	self.loginViewPanelPrefab = Resources.Load("Prefabs/UI/LoginView")
	self.loginViewPanel = GameObject.Instantiate(self.loginViewPanelPrefab,global.UIManager.UICanvas.transform)
	self.usernameInputField = GameObject.Find("UsernameLabel/UsernameInputField"):GetComponent("InputField")
	self.passwordInputField = GameObject.Find("PasswordLabel/PasswordInputField"):GetComponent("InputField")
	self.loginButton = GameObject.Find("LoginButton"):GetComponent("Button")

	self.loginButton.onClick:AddListener(packFunction(self, self.OnclickLoginBtn))

end

function loginView:OnclickLoginBtn( )
	self.username = self.usernameInputField.text
	self.pwd = self.passwordInputField.text

	global.loginController:login()
end

function loginView:destroy( )
	GameObject.Destroy(self.loginViewPanel)
end

return loginView